src/fixit/util.py

```diff
@@ -87,3 +88,36 @@ def append_sys_path(path: Path) -> Generator[None, None, None]:
     # already there: do nothing, and don't remove it later
     else:
         yield
+
+
+class Debouncer:
+    def __init__(self, f: Callable[..., Any], interval: float):
+        self.f = f
+        self.interval = interval
+        self._timer: Optional[threading.Timer] = None
+        self._lock = threading.Lock()
+
+    def __call__(self, *args, **kwargs) -> None:
+        with self._lock:
+            if self._timer is not None:
+                self._timer.cancel()
+            self._timer = threading.Timer(self.interval, self.f, args, kwargs)
+            self._timer.start()
+
+
+VoidFunction = TypeVar("VoidFunction", bound=Callable[..., None])
+
+
+def debounce(interval: float):
+    """
+    Wait `interval` seconds before calling `f`, and cancel if called again.
+    The decorated function will return None immediately,
+    ignoring the delayed return value of `f`.
+    """
+
+    def decorator(f: VoidFunction) -> VoidFunction:
+        if interval <= 0:
+            return f
+        return cast(VoidFunction, Debouncer(f, interval))
+
+    return decorator
```

Comment:
I should add some docs to this module to clarify, but the `util` module is meant more for additional helpers for use by clients/frontends. Is the debouncing helper something you expect would be used by something other than the LSP module? Maybe just put it next to the LSP class otherwise?